### DISCUSSION TITLE
Reading NSIDC Sea Ice Edge NetCDF Files

### DISCUSSION THREAD

**Discussion by GwenChen-NOAA** [dtcenter/METplus#1718] [File I/O] [answered]
Dear METplus team,

We tried to open a NSIDC sea ice edge netCDF file using plot_data_plane, but got the following error messages:

>plot_data_plane NSIDC0081_SEAICE_PS_S25km_20220723_v2.0.nc test.ps 'name="F16_ICECON"; level="(0,*,*)";' -v 4
DEBUG 1: Opening data file: NSIDC0081_SEAICE_PS_S25km_20220723_v2.0.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_NcCF".
DEBUG 4: NcCfFile::open() -> parsing units for the time variable "days since 1970-01-01 00:00:00"
DEBUG 4: parse_cf_time_string() -> parsed NetCDF CF convention time unit string "days since 1970-01-01 00:00:00"
DEBUG 4: 		 as a reference time of 19700101_000000 and 86400 second(s) per time step.
DEBUG 4: NcCfFile::open() -> could not extract init time from the "forecast_reference_time" variable.
DEBUG 4: NcCfFile::open() -> could not extract init time from file name.
ERROR  : 
ERROR  : get_nc_var_att_double() -> Cannot get "longitude_of_projection_origin" from crs variable.
ERROR  : 

This file is in a predefined polar stereographic grid.  It's a curvilinear grid that CDO doesn't recognize and cannot be remapped into latlon grid using "cdo remapbil".  This data set is specified as a validation source for the EVS project at EMC, so we would like to find a way to read it into MET/METplus.  I put two sample files at ftp.rap.ucar.edu:/incoming/irap/met_help/chen_data.  Please advise.

Thanks,
Gwen

**Comment by CPKalb** [comment]
Hi Gwen:

The longitude of projection origin is not an attribute of the crs variable which is the specific error message you are getting.  However, I am not sure what the best way to get this data into MET would be and I need to double check with a few people here.  I will get back to you on this.

Christina

**Comment by JohnHalleyGotway** [comment, marked as answer]
@GwenChen-NOAA, I work with @CPKalb on providing support for METplus.

FYI, I wasn't able to run the two NetCDF files you provided through ncdump or ncview. I assume they got corrupted somewhere along the way. But I did find and test this file available publicly here:
```
https://n5eil01u.ecs.nsidc.org/PM/NSIDC-0081.002/2022.07.23/NSIDC0081_SEAICE_PS_N25km_20220723_v2.0.nc
```
I'm pretty sure you've uncovered a bug in MET in the parsing of Polar Stereographic grids from CF-compliant NetCDF files.

The CF-Convention documentation defines Polar Stereographic grids [here](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#polar-stereographic). And this is parsed by the [get_grid_mapping_polar_stereographic()](https://github.com/dtcenter/MET/blob/0cbdd5f97437990e1ab4642d85de277a99242792/met/src/libcode/vx_data2d_nccf/nccf_file.cc#L2005) function in MET.

I see 2 issues in the code:
1. `longitude_of_projection_origin` is NOT required.
2. Either `scale_factor_at_projection_origin` or `standard_parallel` may be specified, but MET is not parsing/interpreting the latter.

This will require a bugfix to patch up the logic.

**Reply by GwenChen-NOAA** [reply]
Thank you both! And apologize for the corrupted files. I'm not sure what happened to the files, but the one you used is from the same source where I downloaded the data. When will this bugfix be available in MET?

Thanks,
Gwen

**Reply by JohnHalleyGotway** [reply]
@GwenChen-NOAA sorry for the late response here. As we discussed in the last NOAA METplus telecon, we've created a bugfix issue (dtcenter/MET#2218) in MET to correct this. @hsoh-u is working on it now, and we plan to include the fix in the next development version for MET, MET-11.0.0-beta3. Looks like we do also plan to include a fix in the main_v10.1 branch so that it could be included in a MET-10.1.3 bugfix release.

We don't have any immediate plans for the date of that bugfix release. But please feel free to comment directly on the dtcenter/MET#2218 issue to let us know if/when it's needed.

I'll go ahead and mark this discussion as being answered and recommend that any future discussion for it be done via comments on dtcenter/MET#2218.

Thanks!
